WalletWasabi.Fluent/ViewModels/Wallets/Home/History/HistoryViewModel.cs

```diff
@@ -274,8 +275,20 @@ private IEnumerable<HistoryItemViewModelBase> GenerateHistoryList(List<Transacti
 			    (i + 1 < txRecordList.Count && !txRecordList[i + 1].IsOwnCoinjoin || // The next item is not CJ so add the group.
 			     i == txRecordList.Count - 1)) // There is no following item in the list so add the group.
 			{
-				cjg.SetBalance(balance);
-				yield return cjg;
+				if (cjg.CoinJoinTransactions.Count == 1)
+				{
+					var singleCjItem = new CoinJoinHistoryItemViewModel(cjg.OrderIndex, cjg.CoinJoinTransactions.First(), _walletViewModel, balance, _updateTrigger)
+					{
+						Label = new SmartLabel("Coinjoin")
```

Comment:
@wieslawsoltes What about this one, shouldn't this get removed also?

Reply:
Ah yes, fixed that one too